ui/guest/src/components/Guest.js

```diff
@@ -1212,6 +1199,27 @@ export function Guest(props) {
 
   }, [modelId, path]);
 
+  useEffect(() => {
+    $(document).on('dragenter', function (e) {
+      if (stateRef.current.common.status === EditingStatus.LISTENING && e.originalEvent.dataTransfer.types.includes('Files')) {
+        e.preventDefault();
+        e.stopPropagation();
+        fn.onDesktopAssetDragStarted(e.originalEvent.dataTransfer.items[0]);
+      }
+      clearTimeout(persistence.dragLeaveTimeout);
+    });
+    $(document).on('dragover', function () {
+      clearTimeout(persistence.dragLeaveTimeout);
+    });
+    $(document).on('dragleave', function () {
+      clearTimeout(persistence.dragLeaveTimeout);
+      persistence.dragLeaveTimeout = setTimeout(() => {
+        clearTimeout(persistence.dragLeaveTimeout);
+        fn.onDragEnd();
+      }, 100);
+    });
```

Comment:
1. You're missing the destruction logic
2. Try the rxjs-way that I added on a prior comment for the AssetsPanel dragenter/dragleave/dragover

Reply:
pending